python/python/ert/ecl/ecl_grid.py

```diff
@@ -713,11 +714,29 @@ def depth( self , active_index = None , global_index = None , ijk = None):
         gi = self.__global_index( ijk = ijk , active_index = active_index , global_index = global_index)
         return self._get_depth(  gi )
 
-    def top( self , i , j ):
+    def top( self , i=None , j=None, active_index=None ):
         """
-        Top of the reservoir; in the column (@i , @j).
+        Top of the reservoir; in the column (@i , @j).  If instead @active_index
+        is specified, will get top of the cell with given active index.
         """
-        return self._get_top( i , j ) 
+        if i is not None and j is not None and active_index is None:
+            return self._get_top( i , j )
+        if i is None and j is None and active_index is not None:
+            return self._get_top1A( active_index)
+        else:
+            raise ValueError('Either specify i and j, or active_index, got i=%s,j=%s,active_index=%s'
+                             % (str(i), str(j), str(active_index)))
+
+    def top_active( self, i, j ):
+        """
+        Top of the active part of the reservoir; in the column (@i , @j).
+        Returns nan if (i,j) column is inactive.
+        """
+        for k in range(self.getNZ()):
+            a_idx = self.get_active_index(ijk=(i,j,k))
+            if a_idx >= 0:
+                return self.top(active_index=a_idx)
+        return float('nan')
```

Comment:
I would prefer if the `top_active()` method called `self._get_top1A( )` directly instead of forwarding to `top( )`. As I see it the forwarding makes `top( )` considerably more cluttered?